app/models/service.rb

```diff
@@ -55,15 +60,31 @@ def generate_secret_config_params
     end
   end
 
-  def git_repo_url_must_use_https
-    begin
-      uri = URI.parse(self.git_repo_url)
-      unless ['https', 'file', ''].include?(uri.scheme)
-        errors.add(:git_repo_url, I18n.t(:not_valid_scheme, scope: [:errors, :service, :git_repo_url]))
+  def check_git_repo_url
+    if git_repo_url && git_repo_url.starts_with?('git@')
+      uri = URI::SshGit.parse(git_repo_url)
+
+      unless uri.user == 'git'
+        errors.add(:git_repo_url, I18n.t(:not_valid_git, scope: [:errors, :service, :git_repo_url]))
       end
 
-    rescue URI::InvalidURIError => e
-      errors.add(:git_repo_url, I18n.t(:invalid_uri, scope: [:errors, :service, :git_repo_url]))
+      unless uri.host == 'github.com'
+        errors.add(:git_repo_url, I18n.t(:not_valid_git, scope: [:errors, :service, :git_repo_url]))
+      end
+
+      unless uri.path.present?
+        errors.add(:git_repo_url, I18n.t(:not_valid_git, scope: [:errors, :service, :git_repo_url]))
+      end
+    else
+      begin
+        uri = URI.parse(self.git_repo_url)
+        unless ['https', 'file', ''].include?(uri.scheme)
+          errors.add(:git_repo_url, I18n.t(:not_valid_scheme, scope: [:errors, :service, :git_repo_url]))
+        end
+
+      rescue URI::InvalidURIError => e
+        errors.add(:git_repo_url, I18n.t(:invalid_uri, scope: [:errors, :service, :git_repo_url]))
+      end
```

Comment:
This method is a monster at 25 lines. We don't have any linting to complain about it, but as a new developer looking at this code for the first time it would be a lot more readable if it was broken down into a few smaller methods with readable names.

Eg:
```
def check_git_repo_url
  return if invalid_git_repo_url
  add_error if uri_user_is_incorrect
  ... 
end
```

Theres also a lot of duplication of
```
errors.add(:git_repo_url, I18n.t(:not_valid_git, scope: [:errors, :service, :git_repo_url]))
```

And theres a lot of use of `unless` which makes it harder to read than using `if`.

Reply:
sure, i've extracted some methods to make it shorter. however these new methods don't really belong in this class so i think at some point in the future this needs to be moved to a custom validator. I'm happy to accept this localised issue for now